### PR TITLE
Create ecosystem spec 

### PR DESCRIPTION
**Pull request by Emily-Jiang**

### CODE REVIEW

**Comment by Emily-Jiang** [comment]
This is to supersede the PR #26

File: spec/src/main/asciidoc/ecosystem.asciidoc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+== Service Mesh usage of Liveness and Readiness
+
+Service Mesh e.g. Istio can utilise the readiness and liveness status from the underline component such as Kubernetes. 
+Kubernetes provides liveness probes or readiness probes to detect and remedy such situations. Kubernetes can check the pods frequently. 
+If the pod is not live, it will destroy the pod and start a new one. If the application is not ready, Kubernetes doesn’t want to kill the application, but not to send it requests either. 
+
+
+== The ecosystem
+Service mesh can utilise the endpoint exposed by MicroProfile Health for its liveness probe, so that it can control whether to destroy the pod or not.
```
Comment (kenfinnigan) [inline spec/src/main/asciidoc/ecosystem.asciidoc]: My understanding was that Service Mesh, in particular Istio, doesn't have it's own liveness probe.

The service liveness is still determined by Kubernetes liveness probe. Being a sidecar Istio is not involved in that at all

**Comment by pilhuhn** [comment]
> My understanding was that Service Mesh, in particular Istio, doesn't 
> have it's own liveness probe.

Yes.

> The service liveness is still determined by Kubernetes liveness probe. 
> Being a sidecar Istio is not involved in that at all

Not exactly. The communication from the kubelet to the target container
goes through the sidecar. If Istio uses mTLS to that pod, then it is
not possible for the kubelet to talk to the health endpoint, at the
kubelet does not participate in mTLS.

**Comment by Emily-Jiang** [comment]
Hope this revised sentence is acceptable!

**Comment by Emily-Jiang** [comment]
@pilhuhn @kenfinnigan pls check again to see whether it is ok now

File: spec/src/main/asciidoc/ecosystem.asciidoc [hunk 2]
```diff
@@ -92,7 +92,7 @@ If the pod is not live, it will destroy the pod and start a new one. If the appl
 
 
 == The ecosystem
-Service mesh can utilise the endpoint exposed by MicroProfile Health for its liveness probe, so that it can control whether to destroy the pod or not.
+The health check of Istio services can be achieved via Kubernetes liveness and readiness probe, where MicroProfile health attributes via the exposed health check endpoints with some meaningful response. The response of the liveness probe determines whether to recycle the corresponding pod, while the readiness response determines whether requests can be routed.
```
Comment (kenfinnigan) [inline spec/src/main/asciidoc/ecosystem.asciidoc]: My only comment is whether the use of "Istio" here is needed?

Comment (Emily-Jiang) [reply spec/src/main/asciidoc/ecosystem.asciidoc]: changed to "services in the mesh" as agreed on today's hangout.

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**